04-onion-routing.md

```diff
@@ -248,6 +248,13 @@ This is a more flexible format, which avoids the redundant `short_channel_id` fi
     1. type: 6 (`short_channel_id`)
     2. data:
         * [`short_channel_id`:`short_channel_id`]
+    1. type: 10 (`option_amp`)
+    2. data:
+        * [`32*byte`:`payment_id`]
```

Comment:
For recurring payments, the idea with AMP is that `payment_id` can be used as a type of 'deposit box', similar to a traditional bank account number. 

But this field is also used to prevent multiple people trying to pay to the same invoice simultaneously from disturbing each others sets.

It seems that both uses of the same field are mutually exclusive?

Reply:
Ok, clarified offline. That is what `set_id` is for